sympy/matrices/expressions/transpose.py

```diff
@@ -37,6 +37,8 @@ def doit(self, **hints):
         if hints.get('deep', True) and isinstance(arg, Basic):
             arg = arg.doit(**hints)
         try:
+            if isinstance(arg, MatrixExpr) and arg.shape == (1, 1):
+                return arg
```

Comment:
If this logic is in `MatrixExpr._eval_transpose` then it should probably be replicated in `MatMul._eval_transpose` and some other places where `_eval_transpose` is overridden.